session/session.go

```diff
@@ -1056,13 +1095,111 @@ func (s *session) SetGlobalSysVar(name, value string) error {
 		return err
 	}
 	name = strings.ToLower(name)
+	// update mysql.tidb if required.
+	if s.varFromTiDBTable(name) {
+		if err = s.setTiDBTableValue(name, sVal); err != nil {
+			return err
+		}
+	}
 	variable.CheckDeprecationSetSystemVar(s.sessionVars, name)
 	sql := fmt.Sprintf(`REPLACE %s.%s VALUES ('%s', '%s');`,
-		mysql.SystemDB, mysql.GlobalVariablesTable, name, sVal)
+		mysql.SystemDB, mysql.GlobalVariablesTable, name, escapeUserString(sVal))
 	_, _, err = s.ExecRestrictedSQL(sql)
 	return err
 }
 
+// escape user supplied string for internal SQL. Not safe for all cases, since it doesn't
+// handle quote-type, sql-mode, character set breakout.
+func escapeUserString(str string) string {
+	return strings.ReplaceAll(str, `'`, `\'`)
+}
```

Comment:
This still doesn't looks safe enough 😕 I think we need better way to do this in the future

Reply:
Yes, I agree. I searched and couldn't find anything reliable that could be imported in `util` or in a library. See https://github.com/golang/go/issues/18478 for example.